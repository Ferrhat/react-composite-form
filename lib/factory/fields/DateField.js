import React, {Component} from "react";
import DatePicker from 'react-datepicker';
import moment from 'moment';



class DateField extends Component {
    render() {
        const { name, label, value, valueUpdateHandler, changeHandler, disabled } = this.props;

        const onChange = (value) => {
            const currentValue = value ? value.format('YYYY-MM-DD') : null;
            valueUpdateHandler(name, currentValue);
            changeHandler(value);
        }
        return (
            <DatePicker
                className="form-control"
                dateFormat="YYYY-MM-DD"
                disabledKeyboardNavigation
                placeholderText={ label }
                selected={ value ? moment(value, "YYYY-MM-DD") : null }
                onChange={ onChange }
                disabled={ disabled }
            />
        );
    }
}

export default DateField;
