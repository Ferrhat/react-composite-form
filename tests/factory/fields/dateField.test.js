import React from 'react';
import {shallow, mount} from 'enzyme';
import DateField from "../../../lib/factory/fields/DateField";
import moment from 'moment';


describe('DateField', () => {
    const mockValueUpdateHandler = jest.fn();
    const mockChangeHandler = jest.fn();
    const dateField = shallow(<DateField valueUpdateHandler={mockValueUpdateHandler} changeHandler={mockChangeHandler} name={'testName'} />);

    it('renders properly', () => {
        expect(dateField).toMatchSnapshot();
    });

    it('contains a connected DatePicker component', () => {
        expect(dateField.find('DatePicker').length).toEqual(1);
    });

    it('has a default value', () => {
        dateField.setProps({value: '2018-12-31'}, () => {
            expect(dateField.find('DatePicker').prop('selected')).toEqual(moment('2018-12-31', 'YYYY-MM-DD'));
        });
    });

    it('should trigger a change event without a valid date', () => {
        dateField.find('DatePicker').simulate('change', null);
        expect(mockValueUpdateHandler).toBeCalledWith('testName', null);
        expect(mockChangeHandler).toBeCalledWith(null);
    });

    it('should trigger a change event with a valid date', () => {
        const value = moment('2018-12-31');
        dateField.find('DatePicker').simulate('change', value);
        expect(mockValueUpdateHandler).toBeCalledWith('testName', '2018-12-31');
        expect(mockChangeHandler).toBeCalledWith(value);
    });

});
