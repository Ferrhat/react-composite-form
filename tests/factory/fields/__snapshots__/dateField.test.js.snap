// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateField renders properly 1`] = `
<div>
  <DatePicker
    allowSameDay={false}
    className="form-control"
    dateFormat="YYYY-MM-DD"
    dateFormatCalendar="MMMM YYYY"
    disabled={false}
    disabledKeyboardNavigation={true}
    dropdownMode="scroll"
    monthsShown={1}
    onBlur={[Function]}
    onChange={[Function]}
    onClickOutside={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMonthChange={[Function]}
    onSelect={[Function]}
    onYearChange={[Function]}
    preventOpenOnFocus={false}
    selected={null}
    shouldCloseOnSelect={true}
    showTimeSelect={false}
    timeCaption="Time"
    timeIntervals={30}
    withPortal={false}
  />
  <div
    className="validationErrorMessage"
  />
</div>
`;
